internal/storage/caching/fast_stat_bucket.go

```diff
@@ -386,16 +381,24 @@ func (b *fastStatBucket) GetFolder(
 		return entry, nil
 	}
 
-	// Fetch the Folder
-	folder, error := b.wrapped.GetFolder(ctx, prefix)
+	// Fetch the Folder from GCS
+	return b.getFolderFromGCS(ctx, prefix)
+}
 
-	if error != nil {
-		return nil, error
+func (b *fastStatBucket) getFolderFromGCS(ctx context.Context, prefix string) (*gcs.Folder, error) {
+	f, err := b.wrapped.GetFolder(ctx, prefix)
+	if err != nil {
```

Comment:
to avoid nested if, please swap the if-else

if err == nil {
b.insertFolder(f)
return f, nil
}

Reply:
Done